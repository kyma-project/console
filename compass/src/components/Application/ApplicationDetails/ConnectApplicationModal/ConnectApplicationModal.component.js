import React from 'react';
import PropTypes from 'prop-types';
import copyToCliboard from 'copy-to-clipboard';
import { Tooltip } from 'react-shared';

import {
  Button,
  FormSet,
  FormItem,
  FormLabel,
  FormInput,
  FormMessage,
} from 'fundamental-react';
import './ConnectApplicationModal.scss';
import { Modal } from 'shared/components/Modal/Modal';

ConnectApplicationModal.propTypes = {
  applicationId: PropTypes.string.isRequired,
  connectApplicationMutation: PropTypes.func.isRequired,
};

const FormEntry = ({ caption, name, value }) => (
  <FormItem>
    <FormLabel htmlFor={name}>{caption}</FormLabel>
    <div className="connect-application__input--copyable">
      <FormInput type="text" id={name} value={value || 'Loading...'} readOnly />
      {value && (
        <Tooltip title="Copy to clipboard" position="top">
          <Button
            option="light"
            glyph="copy"
            onClick={() => copyToCliboard(value)}
          />
        </Tooltip>
      )}
    </div>
  </FormItem>
);

export default function ConnectApplicationModal({
  applicationId,
  connectApplicationMutation,
}) {
  const [error, setError] = React.useState('');
  const [connectionData, setConnectionData] = React.useState({});

  const connectApplication = async id => {
    try {
      const { data } = await connectApplicationMutation(id);
      setConnectionData(data.requestOneTimeTokenForApplication);
    } catch (e) {
      console.warn(e);
      setError(e.message || 'Error!');
    }
  };

  const clearConnectionData = () => {
    setConnectionData({}); // reset token
  };

  const modalContent = error ? (
    <FormMessage type="error">{error}</FormMessage>
  ) : (
    <FormSet>
      <FormEntry caption="Token" name="token" value={connectionData.token} />
      <FormEntry
        caption="Connector URL"
        name="connector-url"
        value={connectionData.connectorURL}
      />
    </FormSet>
  );

  return (
    <>
      <Modal
        modalOpeningComponent={
          <Button
            option="emphasized"
            onClick={() => connectApplication(applicationId)}
            data-test-id="open-modal"
          >
            Connect Application
          </Button>
        }
        title="Connect Application"
        confirmText="Close"
        onHide={clearConnectionData}
      >
        <section className="connect-application__content">
          {modalContent}
        </section>
      </Modal>
    </>
  );
}
