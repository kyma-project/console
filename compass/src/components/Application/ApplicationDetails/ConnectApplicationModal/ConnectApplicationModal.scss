.form-entry--copyable {
    display: flex;
    justify-content: space-between;

    input {
        width: 24em !important; // to override .fd-input
        margin-right: 16px;
    }
}