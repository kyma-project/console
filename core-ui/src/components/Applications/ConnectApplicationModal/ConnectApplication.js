import React from 'react';
import PropTypes from 'prop-types';
import { CopiableText } from 'react-shared';
import {
  FormSet,
  FormItem,
  FormLabel,
  FormTextarea,
  FormMessage,
} from 'fundamental-react';
import './ConnectApplication.scss';
import { CompassGqlContext } from 'index';

import { useMutation } from '@apollo/react-hooks';
import { CONNECT_APPLICATION } from 'gql/mutations';
const FormEntry = ({ caption, name, value }) => (
  <FormItem className="connect-application__data-entry">
    <FormLabel htmlFor={name}>{caption}</FormLabel>
    <CopiableText textToCopy={value || ''}>
      <FormTextarea
        type="text"
        id={name}
        value={value || 'Loading...'}
        readOnly
      />
    </CopiableText>
  </FormItem>
);

ConnectApplication.propTypes = {
  applicationId: PropTypes.string.isRequired,
};

export default function ConnectApplication({ applicationId }) {
  const compassGqlClient = React.useContext(CompassGqlContext);
  const [connectApplicationMutation] = useMutation(CONNECT_APPLICATION, {
    client: compassGqlClient,
  });

  const [error, setError] = React.useState('');
  const [connectionData, setConnectionData] = React.useState({});

  React.useEffect(() => {
    async function connectApplication(id) {
      try {
        const { data } = await connectApplicationMutation({
          variables: { id },
        });
        setConnectionData(data.requestOneTimeTokenForApplication);
      } catch (e) {
        console.warn(e);
        setError(e.message || 'Error!');
      }
    }

    connectApplication(applicationId);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const content = error ? (
    <FormMessage type="error">{error}</FormMessage>
  ) : (
    <FormSet>
      <FormEntry
        caption="Data to connect Application (base64 encoded)"
        name="raw-encoded"
        value={connectionData.rawEncoded}
      />
      <FormEntry
        caption="Legacy connector URL"
        name="legacy-connector-url"
        value={connectionData.legacyConnectorURL}
      />
    </FormSet>
  );

  return <section className="connect-application__content">{content}</section>;
}
