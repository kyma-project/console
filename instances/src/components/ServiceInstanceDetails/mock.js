export const Mock = {
  __typename: 'clusterDocsTopic',
  name: 'kyma',
  groupName: 'root',
  displayName: 'Kyma',
  description: 'Overall documentation for Kyma',
  assets: [
    {
      __typename: 'clusterAsset',
      name: 'asset1',
      type: 'markdown',
      files: [
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/01-01-in-a-nutshell.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'In a nutshell',
            type: 'Overview',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-01-overview.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'Overview',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-03-cluster-installation.md',
          metadata: {
            __typename: 'fileMetadata',
            // title: 'Install Kyma on a cluster',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/12-01-try-out-kyma.md',
          metadata: {
            __typename: 'fileMetadata',
            type: 'Examples',
            title: 'Try out Kyma',
          },
        },
      ],
    },
  ],
};
