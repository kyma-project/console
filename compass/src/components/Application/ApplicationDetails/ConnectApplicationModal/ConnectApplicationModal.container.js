import { graphql } from 'react-apollo';
import { compose } from 'recompose';

import { CONNECT_APPLICATION } from '../../gql';

import ConnectApplicationModal from './ConnectApplicationModal.component';

export default compose(
  graphql(CONNECT_APPLICATION, {
    props: ({ mutate }) => ({
      connectApplicationMutation: id =>
        mutate({
          variables: {
            id,
          },
        }),
    }),
  }),
)(ConnectApplicationModal);
