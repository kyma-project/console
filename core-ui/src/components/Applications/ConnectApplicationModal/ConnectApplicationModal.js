import React from 'react';
import PropTypes from 'prop-types';
import { Modal, CopiableText } from 'react-shared';
import {
  Button,
  FormSet,
  FormItem,
  FormLabel,
  FormTextarea,
  FormMessage,
} from 'fundamental-react';
import './ConnectApplicationModal.scss';
import { CompassGqlContext } from 'index';

import { useMutation } from '@apollo/react-hooks';
import { CONNECT_APPLICATION } from 'gql/mutations';

ConnectApplicationModal.propTypes = {
  applicationId: PropTypes.string.isRequired,
};

const FormEntry = ({ caption, name, value }) => (
  <FormItem className="connect-application__data-entry">
    <FormLabel htmlFor={name}>{caption}</FormLabel>
    <CopiableText
      textToCopy={value || ''}
      caption={
        <FormTextarea
          type="text"
          id={name}
          value={value || 'Loading...'}
          readOnly
        />
      }
    />
  </FormItem>
);

export default function ConnectApplicationModal({ applicationId }) {
  const compassGqlClient = React.useContext(CompassGqlContext);
  const [connectApplicationMutation] = useMutation(CONNECT_APPLICATION, {
    client: compassGqlClient,
  });

  const [error, setError] = React.useState('');
  const [connectionData, setConnectionData] = React.useState({});

  async function connectApplication(id) {
    try {
      const { data } = await connectApplicationMutation({ variables: { id } });
      setConnectionData(data.requestOneTimeTokenForApplication);
    } catch (e) {
      console.warn(e);
      setError(e.message || 'Error!');
    }
  }

  function clearConnectionData() {
    setConnectionData({}); // reset token
  }

  const modalContent = error ? (
    <FormMessage type="error">{error}</FormMessage>
  ) : (
    <FormSet>
      <FormEntry
        caption="Data to connect Application (base64 encoded)"
        name="raw-encoded"
        value={connectionData.rawEncoded}
      />
      <FormEntry
        caption="Legacy connector URL"
        name="legacy-connector-url"
        value={connectionData.legacyConnectorURL}
      />
    </FormSet>
  );

  return (
    <>
      <Modal
        modalOpeningComponent={
          <Button
            option="emphasized"
            onClick={() => connectApplication(applicationId)}
          >
            Connect
          </Button>
        }
        title="Connect Application"
        confirmText="Close"
        onHide={clearConnectionData}
      >
        <section className="connect-application__content">
          {modalContent}
        </section>
      </Modal>
    </>
  );
}
