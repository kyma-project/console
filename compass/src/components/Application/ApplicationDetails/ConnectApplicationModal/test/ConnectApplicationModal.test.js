import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { mount } from 'enzyme';
import { validMock, errorMock } from './mock';
import { Modal } from 'fundamental-react';
import { act } from 'react-dom/test-utils';

import ConnectApplicationModal from '../ConnectApplicationModal.container';

describe('ConnectApplicationModal Container', () => {
  //for "Warning: componentWillReceiveProps has been renamed"
  console.warn = jest.fn();

  afterEach(() => {
    console.warn.mockReset();
  });

  afterAll(() => {
    if (console.warn.mock.calls.length) {
      expect(console.warn.mock.calls[0][0]).toMatchSnapshot();
    }
  });

  const openModal = async component =>
    await component
      .findWhere(
        t => t.type() == 'button' && t.text() === 'Connect Application',
      )
      .simulate('click');

  it('Modal is initially closed and opens after button click', async () => {
    const component = mount(
      <MockedProvider addTypename={false} mocks={validMock}>
        <ConnectApplicationModal applicationId="app-id" />
      </MockedProvider>,
    );

    expect(component.find(Modal).getDOMNode()).toBeFalsy();

    await openModal(component);

    expect(component.find(Modal).getDOMNode()).toBeTruthy();
  });

  it('Modal is in "loading" state after open', async () => {
    const component = mount(
      <MockedProvider addTypename={false} mocks={validMock}>
        <ConnectApplicationModal applicationId="app-id" />
      </MockedProvider>,
    );

    await openModal(component);

    const inputs = component
      .findWhere(t => t.type() === 'input')
      .map(i => i.instance().value);
    expect(inputs).toMatchSnapshot();
  });
});
