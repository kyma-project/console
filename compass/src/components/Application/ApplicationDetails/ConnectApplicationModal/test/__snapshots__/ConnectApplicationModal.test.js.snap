// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectApplicationModal Container Modal is in "loading" state after open 1`] = `
Array [
  "Loading...",
  "Loading...",
]
`;
